### PR TITLE
[RISCV][MC] Support imm symbol in parseCSRSystemRegister

### PR DESCRIPTION
**Pull request by zqb-all**

### CODE REVIEW

File: llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp [hunk 1]
```diff
@@ -1891,6 +1894,47 @@ ParseStatus RISCVAsmParser::parseCSRSystemRegister(OperandVector &Operands) {
     if (getParser().parseExpression(Res))
       return ParseStatus::Failure;
 
+    if (Kind == AsmToken::Identifier) {
+      auto SRE = dyn_cast<MCSymbolRefExpr>(Res);
```
[the remark this answers is not in the document]
Comment (zqb-all) [reply llvm/lib/Target/RISCV/AsmParser/RISCVAsmParser.cpp]: done, thanks

**Review by arichardson — COMMENTED** [review]
I did something similar in https://github.com/llvm/llvm-project/pull/67377 but forgot about this change so didn't update it. It might be a good idea to see if you can reuse any of the tests there.

**Comment by zqb-all** [comment]
> I did something similar in #67377 but forgot about this change so didn't update it. It might be a good idea to see if you can reuse any of the tests there.

Thanks!  I found that my implementation would not preferentially use the CSR name when the `.set` and CSR name conflicted.
So eventually I used your implementation method in the https://github.com/llvm/llvm-project/pull/67377.

**Review by arichardson — COMMENTED** [review]
This looks good to me. Since this uses some of my code could you please add `Co-authored-by: Alex Richardson <alexrichardson@google.com>` to the commit message.

**Comment by zqb-all** [comment]
> This looks good to me. Since this uses some of my code could you please add `Co-authored-by: Alex Richardson <alexrichardson@google.com>` to the commit message.

Oh, of course. Sorry I didn't add it before, it's now added, thanks again for the code and review.

**Comment by zqb-all** [comment]
ping

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**